### DISCUSSION TITLE
Breakpoints are getting rejected when starting debugging with netcoredbg

### DISCUSSION THREAD

**Discussion by acikgozb** [mfussenegger/nvim-dap#1156] [Q&A]
Hello, I'm trying to setup a basic debugging experience for C# by using `nvim-dap` and `netcoredbg`.

I installed `netcoredbg` via Mason, and added it to my dap config like this:

```
		local home_path = os.getenv("HOME")
		dap.adapters.coreclr = {
			type = "executable",
			command = home_path .. "/.local/share/nvim/mason/bin/netcoredbg",
			args = { "--interpreter=vscode" },
		}

		dap.configurations.cs = {
			{
				type = "coreclr",
				name = "launch - netcoredbg",
				request = "launch",
				program = function()
					if vim.fn.confirm("Should I recompile first?", "&yes\n&no", 2) == 1 then
						vim.g.dotnet_build_project()
					end
					return vim.g.dotnet_get_dll_path()
				end,
			},
		}
```

The functions `dotnet_build_project` and `dotnet_get_dll_path` are taken from the cookbook.

When I toggle a breakpoint and then start debugging via `continue()`, I see breakpoint letters changing to R from B.

I checked the logs from dap and saw that my breakpoints are not verified:

```
[ DEBUG ] 2024-02-12T09:52:17Z+0300 ] ...gozb/.local/share/nvim/lazy/nvim-dap/lua/dap/session.lua:1676 ]	"request"	{
  arguments = {
    name = "launch - netcoredbg",
    program = "/Users/acikgozb/personal-projects/net-web-api-playground/NetWebApi/bin/Debug/net8.0/NetWebApi.dll",
    request = "launch",
    type = "coreclr"
  },
  command = "launch",
  seq = 3,
  type = "request"
}
[ DEBUG ] 2024-02-12T09:52:17Z+0300 ] ...gozb/.local/share/nvim/lazy/nvim-dap/lua/dap/session.lua:950 ]	1	{
  body = {
    breakpoints = { {
        id = 1,
        line = 18,
        message = "The breakpoint is pending and will be resolved when debugging starts.",
        verified = false
      } }
  },
  command = "setBreakpoints",
  request_seq = 2,
  seq = "4",
  success = true,
  type = "response"
}
[ INFO ] 2024-02-12T09:52:17Z+0300 ] ...gozb/.local/share/nvim/lazy/nvim-dap/lua/dap/session.lua:888 ]	"Breakpoint unverified"	{
  id = 1,
  line = 18,
  message = "The breakpoint is pending and will be resolved when debugging starts.",
  verified = false
}
```

Probably I'm missing something really simple in my configuration but I couldn't find out. Any help would be really appreciated.

**Comment by Rimann91** [comment]
DId you ever figure this out? Im having the exact same issue

**Comment by hockic** [comment]
@Rimann91 

try this if you are on windows
```
vim.opt.shellslash = false
vim.defer_fn(function()
  vim.opt.shellslash = false
end, 5000)
```

setting shellslash to false ensures that neovim uses the platform specific path separator i.e. backslash on windows -> which netcoredbg expects when working with paths
setting shellslash to true converts backslashes to forward slashes which will cause breakpoints to be rejected on some setups

some related issues
- https://github.com/mfussenegger/nvim-dap/issues/1337
- https://github.com/Samsung/netcoredbg/issues/40

**Reply by immanonner** [reply]
Thank you for helping me with a somewhat related problem using nvim-jdtls and its debug adapters! I threw that setting at the top of my ftplugin\java.lua file and my breakpoints were finally recognized (not auto rejected) by nvim-dap.

**Reply by nsevonikola** [reply]
Works like a charm! And thanks @immanonner for referencing nvim-jdtls. Having the same issue in windows.  I setup my config like this in order to make it work on mac/linux:


``` lua
-- Determine OS
if vim.fn.has("mac") == 1 then
	system_os = "mac"
elseif vim.fn.has("unix") == 1 then
	system_os = "linux"
elseif vim.fn.has("win32") == 1 or vim.fn.has("win64") == 1 then
	system_os = "win"
	vim.opt.shellslash = false -- Enable shellslash for Windows compatibility
	vim.defer_fn(function()
		vim.opt.shellslash = false
	end, 5000)
else
	print("OS not found, defaulting to 'linux'")
	system_os = "linux"
end
```

**Comment by vec715** [comment]
does not work in my case, wow

**Comment by vec715** [comment]
and there are still no updates/fixes for this problem, looks like it's impossible to debug dotnet projects via neovim currently, so sad
